magic/magic1.cpp

```diff
@@ -2309,6 +2331,43 @@ bool Creature::isMageLich() {
     return(true);
 }
 
+//*********************************************************************
+//                      doFreeAction
+//*********************************************************************
+// Clears all hold/movement hindering effects
+void Creature::doFreeAction() {
+    if (isEffected("slow"))
+        removeEffect("slow");
+    
+    if (isEffected("hold-person"))
```

Comment:
Looks like `removeEffect` already does these checks to see if the effect is present before removing it

Reply:
Completed as suggested.